pkg/tagset/tags.go

```diff
@@ -0,0 +1,154 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.Datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package tagset
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"github.com/twmb/murmur3"
+)
+
+// Tags contains a set of tags, along with a 64-bit hash considered to be
+// unique to that set of tags.  Tags in the tagset are unique.  The order of
```

Comment:
> Tags in the tagset are unique.

Nothing is enforcing it, I would not detail it as a characteristic of the struct this way.

Reply:
(discussed via voice) It's true that the caller can break this invariant, but only by failing to follow the requirements in the docstrings -- in particular, but calling `DisjointUnion` with sets that are not actually disjoint.  So we'll leave this invariant in place, but I'll improve the docstring for DisjointUnion, and rename it to UnsafeDisjointUnion.